tests/infrastructure/src/main/java/org/mule/test/infrastructure/profiling/tracing/SpanTestHierarchy.java

```diff
@@ -47,15 +48,33 @@ public class SpanTestHierarchy {
   private SpanNode currentNode;
   private SpanNode lastChild;
   private static final String NO_PARENT_SPAN = "0000000000000000";
-  private final HashSet<String> visitedSpans = new HashSet();
+  private final HashSet<String> visitedSpans = new HashSet<>();
   private final HashMap<String, CapturedExportedSpan> spanHashMap = new HashMap<>();
-  private final Collection<CapturedExportedSpan> actualExportedSpans;
+  private Collection<CapturedExportedSpan> actualExportedSpans;
 
   public SpanTestHierarchy(Collection<CapturedExportedSpan> actualExportedSpans) {
     this.actualExportedSpans = actualExportedSpans;
     actualExportedSpans.forEach(span -> spanHashMap.put(span.getSpanId(), span));
   }
 
+  public SpanTestHierarchy() {
```

Comment:
Are these changes supposed to go with this PR or instead with the one corresponding to [W-13103997](https://gus.lightning.force.com/a07EE00001QEBtlYAH)?

Reply:
TLDR: This one is a change that allows a more flexible usage of the SpanTestHierarchy that I was needing to simplify the MuleSdkConnectionHandlingOpenTelemetryTracingTestCase , but does not change/add any assertion as W-13103977 will.

Detailed explanation:

1) It decouples the SpanTestHierarchy from the captured Spans, allowing it's instantiation prior to run the test (now it can be statically instantiated as you can see in MuleSdkConnectionHandlingOpenTelemetryTracingTestCase). This is particularly useful for parametrized tests.

2) It provides a size() function, allowing to assert on that value instead of hardcoding the size of the hierarchy.

3) Small change to assert the same hierarchy over different taces via .withCapturedSpans().assertSpanTree()